src/app/public/src/modules/clipboard/clipboard.directive.ts

```diff
@@ -0,0 +1,80 @@
+// adapted from https://github.com/maxisam/ngx-clipboard/
+import {
+  Directive,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  Renderer2,
+  ElementRef,
+  AfterViewInit,
+  ChangeDetectorRef,
+  Input
+} from '@angular/core';
+
+import { StacheClipboardService } from './clipboard.service';
+
+@Directive({
+  selector: '[stacheClipboard]'
+})
+export class StacheClipboardDirective implements AfterViewInit, OnDestroy {
+  // --------------------------------------------------------------
+  // NOTE: Only supported for Code-Block component as of July 2018
+  // --------------------------------------------------------------
+
+  private defaultTigger = 'clipboardButton';
+
+  @Input() public clipboardTrigger: string = this.defaultTigger;
+  @Input() public copySuccessText: string;
+  @Output() public cbOnSuccess: EventEmitter<any> = new EventEmitter<any>();
+  @Output() public cbOnError: EventEmitter<any> = new EventEmitter<any>();
+
+  private targetElm: any;
+  private button: HTMLButtonElement;
+
+  constructor(
+    private clipboardSrv: StacheClipboardService,
+    private cdr: ChangeDetectorRef,
+    private el: ElementRef,
+    private renderer: Renderer2
+  ) { }
+
+  public ngAfterViewInit() {
+    this.targetElm = this.el.nativeElement;
+    this.instantiateButton();
+    this.createEventListeners();
+    this.cdr.detectChanges();
+  }
+
+  public ngOnDestroy() {
+    this.clipboardSrv.destroy();
```

Comment:
Angular services can implement `ngOnDestroy`, so this would be redundant. Make sure you add `StacheClipboardService` to this directive's providers!

See: https://stackoverflow.com/questions/45898948/angular-4-ngondestroy-in-service-destroy-observable